timesketch/frontend/src/components/Sigma/SigmaList.vue

```diff
@@ -127,14 +186,167 @@ export default {
       return this.$store.state.meta
     },
   },
+  created() {
+    if (this.timeline) {
+      ApiClient.getSketchTimelineAnalysis(this.sketch.id, this.timeline.id)
+        .then(response => {
+          this.analyses = response.data.objects[0]
+        })
+        .catch(e => { })
+    }
+    // If no timeline was specified then loop over all of them
+    else {
+      this.sketch.timelines.forEach(timeline => {
+        ApiClient.getSketchTimelineAnalysis(this.sketch.id, timeline.id)
+          .then(response => {
+            this.analyses = this.analyses.concat(response.data.objects[0])
+          })
+          .catch(e => { })
+      })
+    }
+  },
+  mounted() {
+    this.loadSketchSigmaTags()
+    this.loadSketchTTP()
+  },
   methods: {
-    parseSigma: function(event) {
-      ApiClient.getSigmaByText(this.text)
+    // Set debounce to 300ms if parseSigma is used.
+    parseSigma: _.debounce(function (rule_yaml) { // eslint-disable-line
+      this.parsing_issues = []
+      ApiClient.getSigmaRuleByText(rule_yaml)
         .then(response => {
           let SigmaRule = response.data.objects[0]
           this.parsed = SigmaRule
         })
-        .catch(e => {})
+        .catch(e => {
+          Snackbar.open({
+            message: 'Sigma rule parsing failed. See Browser console for more',
+            type: 'is-danger',
+            position: 'is-top',
+            indefinite: false,
+          })
+        })
+    }, 300),
+
+
+
+
+
+    addRule: function (event) {
+      if (this.save_button_text === "Create") {
+        ApiClient.getSigmaRuleByText(this.editingRule.rule_yaml)
+          .then(response => {
+            let SigmaRule = response.data.objects[0]
+            this.parsed = SigmaRule
+            ApiClient.createSigmaRule(this.editingRule.rule_yaml).then(response => {
+              this.$buefy.notification.open({ message: 'Succesfully added Sigma rule!', type: 'is-success' })
+              this.showEditModal = false
+              this.sigmaRuleList.push(response.data.objects[0])
+            })
+              .catch(e => {
+                console.error(e)
+              })
+          })
+          .catch(e => {
+          })
+      }
+      if (this.save_button_text === "Update") {
+        // Only update the rule if the parsing was positive.
+        ApiClient.getSigmaRuleByText(this.editingRule.rule_yaml)
+          .then(response => {
+            let SigmaRule = response.data.objects[0]
+            this.parsed = SigmaRule
+            ApiClient.updateSigmaRule(this.editingRule.id, this.editingRule.rule_yaml)
+              .then(response => {
+                this.$store.state.sigmaRuleList = this.sigmaRuleList.filter(obj => {
+                  return obj.rule_uuid !== this.editingRule.rule_uuid
+                })
+                this.sigmaRuleList.push(response.data.objects[0])
+              })
+              .catch(e => {
+                console.error(e)
+              })
```

Comment:
Can we make this error message more apparent to the user?

Reply:
https://github.com/google/timesketch/issues/2357